agent/src/api/mod.rs

```diff
@@ -86,36 +86,16 @@ pub struct Envelope {
 /// Remember what is important.
 pub async fn clacks<B>(req: Request<B>, next: Next<B>) -> Result<Response, StatusCode> {
     let mut response = next.run(req).await;
-    response.headers_mut().append(
-        "X-Clacks-Overhead",
-        HeaderValue::from_static("GNU/Terry Pratchett"),
-    );
+    if !response.headers().contains_key("X-Clacks-Overhead") {
```

Comment:
Prior to this, proxied requests were ending up with two copies of this in their responses (one from each node that was involved). It's possible that I could've solved this else-how by swapping the order of the two `.route_layer` calls that install our middleware, but I am not sure that's any more grokkable. Happy to do this elsehow if you prefer.